protocol/git/outdated.md

```diff
@@ -0,0 +1,120 @@
+Git Protocol
```

Comment:
I understand the desire to keep this PR minimal in scope, but it seems like a net-negative to hide some documentation in a file named `outdated.md` :)

Thoughts on deciding to:

1. Totally remove the Git protocol file, or;
2. Moving some of it that seems relevant into `git/README.md` and deleting the rest, or;
3. Keeping both files around, and rename the file with the new content to `deployment_pipeline.md` or something similar?